### ISSUE TITLE
crystal orientation of slab in all directions

### ISSUE THREAD

**Issue by bipinkumar32** [materialsproject/pymatgen#2386]
I created a slab by SlabGenerator of pymatgen.core.surface. I see the miller_index of the slab surface. However, I also want to know the crystal orientation of other two orthogonal directions. How is it possible by pymatgen?

**Comment by shyuep** [comment]
Not quite sure what you mean. If you are referring to the lattice directions, that can be obtained from the Slab.lattice object.

**Comment by bipinkumar32** [comment]
Ok, maybe my question was not clear.
Suppose that I created a slab that has the crystal direction say [1-100] (for HCP family as an example) of the slab surface. Now, I want to know the two crystallographic directions that are orthogonal to one another and also to [1-100] direction.

**Comment by shyuep** [comment]
The c lattice vector is always the normal to the plane for Slab objects. The plane itself is defined by vectors a and b. Those are not always perpendicular to c. There is no reason why they must be. If you want them to be perpendicular, you can use the get_orthgonal_c_slab method. 

As an aside, these questions should be posed on the matsci.org forum since they are not "issues" with pymatgen per se.

**Comment by bipinkumar32** [comment]
As I understood, slab.miller_index provides the crystallographic direction along the c vector. 
Yes, a and b do not have to be orthogonal to create a slab and just need to be normal to c vector. However, c vector does not necessarily always represent the crystallographic direction of the surface. A crystallographic orientation of a slab surface, say (1-10), can be along the c vector that is [0 0 c_z]. I wonder if it is possible to find (by pymatgen) the crystallographic direction along a and b vectors even if they are not orthogonal. 

In other words, as I can find miller index of the slab surface (which is normal to c vector), can I also find similar along a and b vectors?

Thanks for the link.
